### ISSUE TITLE
Investigate if there are deps/assets we can leave uncompressed to improve performance

### ISSUE THREAD

**Issue by mcomella** [mozilla-mobile/fenix#9382]
In https://github.com/mozilla-mobile/fenix/pull/9363/, we identified that compressing `omni.ja` slows down performance. What other dependencies/assets might we leave uncompressed to improve performance?

@snorp Do you have any ideas?

**Comment by mcomella** [comment]
@jeanygong Can you:
- Follow-up acreskey on whether the `omni.ja` change was impactful and if it's worth applying similar changes to other native dependencies
- If so, file a bug against AppServices, Glean, and GeckoView to see if they can apply similar changes

**Comment by snorp** [comment]
Native libraries already get extracted to the filesystem upon install, so there shouldn't be anything to gain there performance-wise.

We could also consider not compressing some (but probably not all) things in the omni.ja itself, but AFAIK profiles don't show that being a problem.

**Comment by mcomella** [comment]
@mstange Would it be possible to get this information with:
- The Debug API
- Log statements
- Some other method?

What are the next action items for this bug? Do you agree that we should we confirm the decompression has changed before filing bugs for other teams using native code?

**Comment by mstange** [comment]
I'm not sure.
The Debug API doesn't give us C/C++ stacks, as far as I know, and decompression likely happens in native code. As for log statements, I wouldn't know where to insert them.

The only action item I can think of is: "Get a simpleperf startup profile of a release build."
The [simpleperf instructions](https://android.googlesource.com/platform/system/extras/+/master/simpleperf/doc/android_application_profiling.md#prepare-an-android-application) say that this is possible, either with `<profileable android:shell="true" />` on Android >= Q, or with a wrapper script on Android >= O, or without any trouble on a rooted device.

> Do you agree that we should we confirm the decompression has changed before filing bugs for other teams using native code?

What do you mean by this?

**Comment by acreskeyMoz** [comment]
Let me see if the art::DexFileVerifier::Verify is showing up via `simpleperf`.
Adding this to the manifest looks straight forward enough: `<profileable android:shell="true" />`

**Comment by mcomella** [comment]
@acreskeyMoz Would you mind taking ownership over pushing this bug forward? It seems more like your territory than mine. :)

**Comment by acreskeyMoz** [comment]
@mcomella yes -- I can't assign it to myself (I'm guessing because I'm not a Fenix contributor), but I can keep taking the next steps on it.

**Comment by acreskeyMoz** [comment]
Trying to see if we see decompression in non-debuggable builds.

• I couldn't add ```<profileable android:shell="true" />``` because I don't have >= Q device.

• Using a wrapper script, I was still seeing `art::DexFileVerifier::Verify` on the debuggable APK that's launched without the debug flags to ART.
https://android.googlesource.com/platform/system/extras/+/master/simpleperf/doc/android_application_profiling.md#prepare-an-android-application

• To verify on a non-modified non-debuggable build (e.g. nightly), I attempted to simpleperf it on rooted emulator, however I ran into this problem:
```simpleperf E event_selection_set.cpp:201] Event type 'cpu-cycles' is not supported on the device```

• @MarcLeclair and I are trying this on his rooted device, although so far it's still problematic.

**Comment by acreskeyMoz** [comment]
Without a rooted physical android device, I am blocked on attempting to simpleperf non-debuggable apps.
@MarcLeclair Do you want to take ownership of this bug?
Otherwise we may want to put this on hold until someone is setup for this.

**Comment by acreskeyMoz** [comment]
@MarcLeclair was able to root a device and can successfully capture simplerperf profiles on non-debuggable, unmodified, apps.

I believe that this is the first time we've been able to get a broad and low overhead view of the performance of these shippable builds.

Imported into the gecko profiler format, this is startup to Home Activity:
https://bit.ly/2YeKar6

I only see 1ms of `inflate_fast` in the main java thread.
And 7ms in the Gecko thread.

This leads me to believe that there isn't anything to do here with respect to decompression.

**Comment by mstange** [comment]
> @MarcLeclair was able to root a device and can successfully capture simplerperf profiles on non-debuggable, unmodified, apps.
> 
> I believe that this is the first time we've been able to get a broad and low overhead view of the performance of these shippable builds.

🎉🎉🎉🎉🎉🎉🎉

This is phenomenal.

**Comment by enh-google** [comment]
(apologies for slightly spamming your bug, but...) ... this bug was the only search result for that particular simpleperf error message when i hit it myself.

since i happen to work with the simpleperf author (but like to make sure things are documented publicly) i asked him on https://github.com/android/ndk/issues/1558 for a workaround. he suggested `-e cpu-clock`, which worked for me. (disclaimer: i was on https://source.android.com/setup/create/cuttlefish rather on a regular emulator, but it should work for both.)

he's also said he'll just make simpleperf fall back to cpu-clock when cpu-cycles isn't available, so hopefully on a long enough time frame no more app developers will need to scratch their heads about this. and, in case you have other issues with simpleperf, you can file bugs/feature requests over at https://github.com/android/ndk/issues and we'll make sure they're routed to someone who knows more than i do :-)

we're glad to hear you're finding simpleperf useful!

**Comment by mstange** [comment]
Thanks for the note. This is a useful workaround to have available, for performance issues that reproduce on the emulator.
